hs/src/Reach/IdClient.hs

```diff
@@ -0,0 +1,207 @@
+module Reach.IdClient
+  ( AuthResult(..)
+  , createAccount
+  , defaultEnv
+  , getAccountInfo
+  , login
+  , logout
+  , Password(..)
+  , sendVerificationCode
+  , updateAccount
+  , updateEmail
+  , updatePassword
+  , UserId(..)
+  , Username(..)
+  , VerCode(..)
+  , verifyMfa
+  ) where
+
+import           Control.Applicative
+import           Control.Monad.IO.Class
+import           Control.Monad.Reader
+import           Data.Aeson
+import           Data.ByteString        (ByteString)
+import           Data.IORef
+import           Data.Maybe
+import           Network.HTTP.Client
+import           Network.HTTP.Simple
+import           Reach.AST.ID.Req
+import           Reach.Util
+
```

Comment:
Heresy!

Reply:
you need a better editor to do this formatting for you 😛